micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/metrics/config/GraphitePublisherRegistrationSpec.groovy

```diff
@@ -0,0 +1,56 @@
+package com.ofg.infrastructure.metrics.config
+
+import com.codahale.metrics.graphite.Graphite
+import com.ofg.infrastructure.base.BaseConfiguration
+import com.ofg.infrastructure.metrics.publishing.GraphitePublisher
+import org.springframework.context.annotation.AnnotationConfigApplicationContext
+import org.springframework.mock.env.MockPropertySource
+import spock.lang.AutoCleanup
+import spock.lang.Specification
+
+import static com.ofg.config.BasicProfiles.PRODUCTION
+
+class GraphitePublisherRegistrationSpec extends Specification {
+
+    @AutoCleanup AnnotationConfigApplicationContext applicationContext
+
+    def setup() {
+        applicationContext = new AnnotationConfigApplicationContext()
+        applicationContext.environment.setActiveProfiles(PRODUCTION)
+        applicationContext.register(BaseConfiguration, MetricsConfiguration)
+    }
+
+    def 'should register Graphite beans when graphite-publishing is on'() {
+        given:
+            applicationContext.environment.propertySources.addFirst(developerEnvDisabledPropertySource())
+            applicationContext.refresh()
+        expect:
+            applicationContext.getBean(GraphitePublisher) != null
+            applicationContext.getBean(Graphite) != null
+    }
+
+    def 'should not register Graphite beans when graphite-publishing is off'() {
+        given:
+            applicationContext.environment.propertySources.addFirst(developerEnvEnabledPropertySource())
+            applicationContext.refresh()
+        expect:
+            applicationContext.getBeanNamesForType(GraphitePublisher) == []
+            applicationContext.getBeanNamesForType(Graphite) == []
+    }
+
+    def 'should register Graphite beans when graphite-publishing is not defined'() {
+        given:
+            applicationContext.refresh()
+        expect:
+            applicationContext.getBean(GraphitePublisher) != null
+            applicationContext.getBean(Graphite) != null
+    }
+
+    def developerEnvDisabledPropertySource() {
+        return new MockPropertySource().withProperty('graphite-publishing', 'on')
```

Comment:
Can we support more options, like `true`, `yes`, etc.? Case insensitive? Or support at least boolean, more common